indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/Asn1Elements.kt

```diff
@@ -326,24 +494,25 @@ data class TLV(val tag: UByte, val content: ByteArray) {
 
         other as TLV
 
-        if (tag != other.tag) return false
+        if (tag == other.tag) return false
         if (!content.contentEquals(other.content)) return false
 
         return true
     }
 
     override fun hashCode(): Int {
-        var result = tag.toInt()
+        var result = tag.hashCode()
         result = 31 * result + content.contentHashCode()
         return result
     }
 
     override fun toString(): String {
-        return "TLV(tag=0x${byteArrayOf(tag.toByte()).encodeToString(Base16)}" +
-                ", length=$length" +
+        return "TLV(tag=$tag" +
+                ", length=$contentLength" +
                 ", overallLength=$overallLength" +
                 ", content=${content.encodeToString(Base16)})"
     }
+
 }
 
 @Throws(IllegalArgumentException::class)
```

Comment:
This `encodeLength` now looks like it might not handle all cases ... is there some other function that can be used for encoding?

Reply:
which line(s) does this comment address? GitHub does not render it in a meaningful manner:

<img width="883" alt="image" src="https://github.com/user-attachments/assets/834eea0c-e918-4ccb-abd4-6540f9ddbbf3">

Reply:
It addresses the fact that `encodeLength` throws `IllegalArgumentException`, I would think.

Reply:
The function `encodeLength` starting in line 519

Reply:
we are using it on array lengths which are integers by default and the function is internal. I am a bit confused

Reply:
why does this special-case out particular ranges, and not work as a generic varint encoder

Reply:
squint your eyes really hard at the code ;-)

Reply:
https://pl.kotl.in/yJyIGqHRD

Reply:
be my guest